src/App.jsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from "react";
+import TodoInput from "./components/TodoInput/TodoInput";
+import TodoList from "./components/TodoList/TodoList";
+import Card from "./components/ui/Card";
+
+function App() {
+	const [inputText, setInputText] = useState("");
+	const [todosList, setTodosList] = useState([]);
+
+	// ! Updating name of task when getting edited
+	const updateTodoName = (updatedValue, id) => {
+		const newTasks = todosList.map(todo => (todo.id === id ? { ...todo, name: updatedValue } : todo));
+		setTodosList([...newTasks]);
+	};
+
+	// ! Getting previous todo list from LocalStorage
+	useEffect(() => {
+		if (JSON.parse(localStorage.getItem("storedTasks")))
+			setTodosList([...JSON.parse(localStorage.getItem("storedTasks"))]);
+	}, []);
+
+	// ! Updating current todo list to LocalStorage
+	useEffect(() => {
+		setTimeout(() => {
+			localStorage.setItem("storedTasks", JSON.stringify(todosList));
+		}, 1000);
+	}, [todosList]);
+
+	const onChangeHandler = e => {
+		setInputText(e.target.value);
+	};
+
+	// ! TODO FUNCTIONS !
+
+	const addTodo = () => {
+		if (!inputText) return;
+		const newTodo = {
+			id: Math.floor(Math.random() * 100000 + 1),
+			name: inputText,
+			completed: false,
+			disabled: true,
+		};
+		setTodosList([...todosList, newTodo]);
+		setInputText("");
+	};
+
+	const deleteTodo = id => {
+		let newTasks = todosList.filter(todo => todo.id !== id);
+		setTodosList([...newTasks]);
+	};
+
+	const markTodo = id => {
+		const newTasks = todosList.map(todo =>
```

Comment:
A nie prościej będzie zrobić mapa, w którym sprawdzasz tylko, czy todo ma id równe przekazanemu i wtedy ustawić jego completed na przeciwne poprzedniemu stanowi? Plus po co ten disabled?

Jeszcze może bym prevState do tego wykorzystał, żeby na pewno był to aktualny stan.

Reply:
Disabled po to, żeby nie można było markować tasków, które są w trakcie edycji. Przy prostszej logice na pewno Twoje rozwiazanie byłoby lepsze, ale postanowiłem nieco zakombinować, żeby użytkowo było przyjemniej (tak na moje oko przynajmniej 😄)